woocommerce/woocommerce-gateway-stripe
# Add a rate limit to the configure webhook request

Part of #3195
Fixes #3213

## Changes proposed in this Pull Request:

This PR adds a rate limit to the **(Re)Configure webhooks** button in the admin settings to prevent sending multiple Stripe requests if the merchant keeps clicking the button. 

<p align="center">
<img width="514" alt="Screenshot 2024-06-19 at 1 46 51 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/7d64d2a5-877c-40b0-a287-1cb5a1a83c90">
</br><sup>If you click the <strong>Configure webhooks</strong> button within a minute of the last request, you will see this response.</sup>
</p>


## Testing instructions

1. Checkout this branch and run `npm run build`
2. Go to your Stripe API settings and click the **Configure webhooks** or **reconfigure webhooks** button. 
3. You should see a success notice.
4. Click the button again and you should see the error message regarding the rate limit. 
5. Wait 1 minute and click the button again, the request should be successful now. 

---


**Post merge**
